lib/modules.js

```diff
@@ -126,59 +113,78 @@ async function getLastSeenModuleDistTag ({ module, moduleVersionsDir }) {
   return undefined
 }
 
-async function setLastSeenModuleDistTag ({ module, distTag, moduleVersionsDir }) {
+async function setLastSeenModuleCID ({ module, cid, moduleVersionsDir }) {
   await mkdir(moduleVersionsDir, { recursive: true })
-  await writeFile(join(moduleVersionsDir, module), distTag)
+  await writeFile(join(moduleVersionsDir, module), cid)
 }
 
-export async function updateSourceFiles ({ module, repo, moduleVersionsDir }) {
+export async function updateSourceFiles ({ module, ipnsKey, moduleVersionsDir }) {
   await mkdir(moduleBinaries, { recursive: true })
   const outDir = join(moduleBinaries, module)
 
-  const lastSeenDistTag = await getLastSeenModuleDistTag({ module, moduleVersionsDir })
-  if (lastSeenDistTag !== undefined) {
+  const lastSeenCID = await getLastSeenModuleCID({ module, moduleVersionsDir })
+  if (lastSeenCID !== undefined) {
     // Use `console.error` because with `--json` stdout needs to be JSON only
     console.error(`[${module}]  ⇣ checking for updates`)
   }
 
-  const distTag = await getLatestDistTag(repo)
-  const isUpdate = lastSeenDistTag !== distTag
+  const cid = await getLatestCID(ipnsKey)
+  const isUpdate = lastSeenCID !== cid
   if (!isUpdate) {
     console.error(`[${module}]  ✓ no update available`)
     return isUpdate
   }
 
-  console.error(`[${module}]  ⇣ downloading source files`)
-  const url = `https://github.com/${repo}/archive/refs/tags/${distTag}.tar.gz`
-  const res = await fetch(url, {
-    headers: {
-      ...(authorization ? { authorization } : {})
-    },
-    redirect: 'follow'
-  })
+  const url = `https://${cid}.ipfs.w3s.link?format=car`
+  console.error(`[${module}]  ⇣ downloading source files via ${url}`)
+  const res = await fetch(url)
 
   if (res.status >= 300) {
     throw new Error(
-      `[${module}] Cannot fetch ${module} archive for tag ${distTag}: ${res.status}\n` +
+      `[${module}] Cannot fetch ${module} archive for ${cid}: ${res.status}\n` +
       await res.text()
     )
   }
 
   if (!res.body) {
     throw new Error(
-      `[${module}] Cannot fetch ${module} archive for tag ${distTag}: no response body`
+      `[${module}] Cannot fetch ${module} archive for ${cid}: no response body`
     )
   }
 
   try {
-    // `{ strip: 1}` tells tar to remove the top-level directory (e.g. `mod-peer-checker-v1.0.0`)
-    await pipeline(res.body, gunzip(), tar.extract(outDir, { strip: 1 }))
+    const reader = await CarReader.fromIterable(res.body)
+    const entries = exporter(cid, {
+      async get (blockCid) {
+        const block = await reader.get(blockCid)
+        try {
+          await validateBlock(block)
+        } catch (err) {
+          throw new Error(
+            `Invalid block ${blockCid} of root ${cid}. ${err.stack}`
```

Comment:
The typescript setup disallows passing a second argument to `Error()`

Reply:
I think this is caused by our TypeScript configuration targeting ES2021 - I guess the Error constructor did not have the second argument back then.

https://github.com/filecoin-station/core/blob/1e9bb5e1dc4fa46036f34d66aa56b386a5c03ff9/tsconfig.json#L6-L7

Depending on what's our minimum supported Node.js version, we can bump this to a newer ES standard.

Reply:
According to https://stackoverflow.com/a/67371788, we can't upgrade as long as we're using node 16. But! Node 16 is past EOL so let's update the minimum engine

Reply:
https://github.com/filecoin-station/core/pull/335